pandas/tests/frame/test_combine_concat.py

```diff
@@ -763,3 +763,14 @@ def test_concat_datetime_datetime64_frame(self):
 
         # it works!
         pd.concat([df1, df2_obj])
+
+
+class TestDataFrameUpdate(TestData):
+
+    def test_update_nan(self):
+        # #15593 #15617
+        expected = df1 = DataFrame({'A': [1.0, 2, 3], 'B': date_range('2000', periods=3)})
```

Comment:
sorry, you are right @TomAugspurger . Hmm that is pretty bogus it should have a kwarg for this.